### PR TITLE
{App Service} `az webapp deploy/deployment`: Added checksum header to check file integrity in service, for linux app service

### PR DESCRIPTION
**Pull request by ShilpiRachna1**
**Related command**
`az webapp deploy`
`az webapp deployment source config-zip`

**Description**
For Linux app service, we have been seeing multiple customer issues where the service side logs indicate that the file sent in the deployment request is corrupt. But we do not know the source of these errors - client, network or service.

Adding file checksum to the request header and matching that with the checksum computed on the service side would enable us to identify the root cause of these errors.

This change is only for linux app service and for local file.

**Testing Guide**

Tested the following scenarios using args in launch.json

   1.   `az webapp deploy -g sumali-prod-test-2023 -n dotnet-8-3108 --src-path D:\test\net8.0.zip`
   2.   `az webapp deployment source config-zip -g sumali-prod-test-2023 -n dotnet-8-3108 --src D:\test\net8.0.zip`

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -6990,12 +7002,25 @@ def _get_onedeploy_status_url(params):
...
             with open(os.path.realpath(os.path.expanduser(params.src_path)), 'rb') as fs:
                 body = fs.read()
+                try:
+                    if app_is_linux_webapp:
+                        import hashlib
+                        logger.info("Computing checksum of the file ...")
+                        file_hash = hashlib.sha256(body).hexdigest()
+                        logger.info("Computed checksum for deployment request header x-ms-artifact-checksum %s", file_hash)
+                except Exception as ex: # pylint: disable=broad-except
+                    logger.info("Computing the checksum of the file failed with exception:'%s'", ex)
```
Comment (surenderssm) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:7023]: Given this is common b/w both the workflows, how about moving it to a function?

**Comment by yanzhudd** [comment]
please fix the style check issue: https://dev.azure.com/azclitools/public/_build/results?buildId=176994&view=logs&j=36dd4138-4d53-5e46-00d9-e5cd9744be05&t=1cf3879c-0a42-5ccd-7693-7a4781d739d8

**Review by shubham805 — APPROVED** [review]

**Comment by ShilpiRachna1** [comment]
> please fix the style check issue: https://dev.azure.com/azclitools/public/_build/results?buildId=176994&view=logs&j=36dd4138-4d53-5e46-00d9-e5cd9744be05&t=1cf3879c-0a42-5ccd-7693-7a4781d739d8

Fixed

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -789,11 +789,16 @@ def enable_zip_deploy(cmd, resource_group_name, name, src, timeout=None, slot=No
...
     from azure.cli.core.util import should_disable_connection_verify
-    # Read file content
+    # check if the app is a linux web app
+    app_is_linux_webapp = is_linux_webapp(app)
 
+    # Read file content
     with open(os.path.realpath(os.path.expanduser(src)), 'rb') as fs:
         zip_content = fs.read()
         logger.warning("Starting zip deployment. This operation can take a while to complete ...")
+        if app_is_linux_webapp and track_status is not None and track_status:
```
Comment (sarsharma) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:799]: nit: we can just do `and track_status`

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 3]
```diff
@@ -6990,12 +6995,22 @@ def _get_onedeploy_status_url(params):
 
 def _get_onedeploy_request_body(params):
     import os
+    file_hash = None
+    app_is_linux_webapp = False
```
Comment (sarsharma) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:6999]: we are not getting this value?

**Review by sarsharma — APPROVED** [review]

**Review by surenderssm — APPROVED** [review]
Looks good!

**Review by sarsharma — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**